thoth/prescriptions_refresh/handlers/image_analysis.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+# thoth-prescriptions-refresh
+# Copyright(C) 2021 Francesco Murdaca
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Create prescriptions from container image analysis results."""
+
+import os
+import requests
+import logging
+from itertools import chain
+from typing import Dict, Any, Optional, Tuple
+
+from thoth.python import Pipfile, PipfileLock
+
+from thoth.prescriptions_refresh.prescriptions import Prescriptions
+from .quay.common import get_configured_image_names
+from .quay.common import get_ps_s2i_image_names
+from .quay.common import get_image_containers
+from .quay.common import QUAY_TOKEN
+from .quay.common import QUAY_URL
+
+
+_LOGGER = logging.getLogger(__name__)
+
+_IMAGE_ANAYSIS_PRESCRIPTION_NAME = "thoth_image_analysis.yaml"
+_THOTH_IMAGE_ANALYSIS_WRAP = """\
+  - name: {prescription_name}BaseImageWrap
+    type: wrap
+    should_include:
+      adviser_pipeline: true
+      recommendation_types:
+      - latest
+      - performance
+      - stable
+      - testing
+      runtime_environments:
+        operating_systems:
+        - name: {os_name}
+          version: {os_version}
+        python_version: =={python_version}
+        base_images:
+          not:
+          - {image}
+    match:
+      state:
+        resolved_dependencies:
+        {resolved_dependencies}
+    run:
+      stack_info:
+      - type: INFO
+        message: >-
+          Found predictive stack image that can be used with these dependencies
+        link: {link}
+      advised_manifest_changes:
+      - file: .thoth.yaml
+        patch:
+        - name: base_image
+          value: {link}
+"""
+
+# TODO: Use Isis-API to identify categories or representative packages per topic?
+
+_REPRESENTATIVE_PACKAGES_ML = ["keras", "scikit-learn", "torch", "tensorflow"]
+
+_REPRESENTATIVE_PACKAGES_NLP = [
+    "gensim",
+    "nltk",
+    "spacy",
+    "transformers"
+]
+
+_REPRESENTATIVE_PACKAGES_CV = ["opencv-python", "pillow", "pytesseract", "torchvision"]
+
+_REPRESENTATIVE_PACKAGES = _REPRESENTATIVE_PACKAGES_ML + _REPRESENTATIVE_PACKAGES_NLP + _REPRESENTATIVE_PACKAGES_CV
+
+USER_API_HOST = os.environ["THOTH_USER_API_HOST"]
```

Comment:
Each handler is run as a separate task in argo workflows. `THOTH_USER_API_HOST` is used just by this handler as of now:

```suggestion
USER_API_HOST = os.getenv("THOTH_USER_API_HOST")
```